colossalai/booster/booster.py

```diff
@@ -64,17 +65,20 @@ def __init__(self,
             assert isinstance(
                 plugin, Plugin), f'Expected the argument plugin to be an instance of Plugin, but got {type(plugin)}.'
         self.plugin = plugin
+        logger = get_dist_logger()
 
         # set accelerator
         if self.plugin and self.plugin.control_device():
             self.accelerator = None
-            warnings.warn('The plugin will control the accelerator, so the device argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
+                           ranks=[0])
         else:
             self.accelerator = Accelerator(device)
 
         # set precision
         if self.plugin and self.plugin.control_precision():
-            warnings.warn('The plugin will control the precision, so the mixed_precision argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
```

Comment:
Only warn when `mixed_precision` is not None

Reply:
modified